core/rawdb/schema.go

```diff
@@ -387,3 +390,58 @@ func utxoToBlockHeightKey(txHash common.Hash, index uint16) []byte {
 	txHash[common.HashLength-2] = indexBytes[1]
 	return append(utxoToBlockHeightPrefix, txHash[:]...)
 }
+
+var CoinbaseLockupKeyLength = len(CoinbaseLockupPrefix) + 2*common.AddressLength + 1 + 4
+
+func CoinbaseLockupKey(ownerContract common.Address, beneficiaryMiner common.Address, lockupByte byte, epoch uint32) []byte {
+	epochBytes := make([]byte, 4)
+	binary.BigEndian.PutUint32(epochBytes, epoch)
+	ownerBytes := ownerContract.Bytes()
+	beneficiaryBytes := beneficiaryMiner.Bytes()
+	combined := append(ownerBytes, beneficiaryBytes...)
+	combined = append(combined, lockupByte)
+	combined = append(combined, epochBytes...)
+	return append(CoinbaseLockupPrefix, combined...)
+}
+
+func createdCoinbaseLockupsKey(hash common.Hash) []byte {
+	return append(createdCoinbaseLockupsPrefix, hash.Bytes()...)
+}
+
+func deletedCoinbaseLockupsKey(hash common.Hash) []byte {
+	return append(deletedCoinbaseLockupsPrefix, hash.Bytes()...)
+}
+
+func ReverseCoinbaseLockupKey(data []byte, location common.Location) (common.Address, common.Address, byte, uint32, error) {
+
+	epochLength := 4 // Length of the epoch in bytes
+	prefixLength := len(CoinbaseLockupPrefix)
+
+	// Ensure the data is long enough to contain all components
+	if len(data) < prefixLength+2*common.AddressLength+1+epochLength {
+		return common.Address{}, common.Address{}, 0, 0, fmt.Errorf("key is too short to parse")
+	}
+
```

Comment:
maybe also add a check to make sure the data is not greater than the max possible key length

Reply:
What is the max possible key length?